templates/project/Makefile

```diff
@@ -1,27 +1,64 @@
 .PHONY: build
 
+# Older ESP32 devices might only have 4mb of flash and slower
+FLASHSIZE:=8mb
+BAUDRATE:=460800
+
+clean:
+	cargo clean
+
 build:
 	cargo build --release
 
 upload:
-	cargo espflash flash --erase-parts nvs --monitor --partition-table partitions.csv --baud 460800 -f 80mhz --release $(ESPFLASH_FLASH_ARGS)
+	cargo espflash flash \
+		--erase-parts nvs \
+		--monitor \
+		--partition-table partitions.csv \
+		--baud ${BAUDRATE} \
+		--flash-freq 80mhz \
+		--release $(ESPFLASH_FLASH_ARGS)
+
+monitor:
+	espflash monitor --baud ${BAUDRATE}
+
+info:
+	cargo espflash board-info --baud ${BAUDRATE}
+
+erase:
+	cargo espflash erase-flash --baud ${BAUDRATE}
 
+list-partitions:
+	cargo espflash partition-table partitions.csv
 
 build-esp32-bin:
-	cargo espflash save-image --merge --chip esp32 target/xtensa-esp32-espidf/release/{{project-name}}.bin --partition-table partitions.csv -s 8mb  --release
+	cargo espflash save-image \
+		--merge \
+		--chip esp32 \
+		--partition-table partitions.csv \
+		--target xtensa-esp32-espidf \
+		--flash-size ${FLASHSIZE} \
+		--release \
+		target/xtensa-esp32-espidf/release/{{project-name}}.bin
 
 build-esp32-ota:
 	cargo +esp espflash save-image \
 		--skip-update-check \
 		--chip=esp32 \
 		--partition-table=partitions.csv \
 		--target=xtensa-esp32-espidf \
-		-Zbuild-std=std,panic_abort --release \
+		-Zbuild-std=std,panic_abort \
+		--release \
 		target/xtensa-esp32-espidf/project-ota.bin
 
 flash-esp32-bin:
 ifneq (,$(wildcard target/xtensa-esp32-espidf/release/{{project-name}}))
-	espflash flash --erase-parts nvs --partition-table partitions.csv  target/xtensa-esp32-espidf/release/{{project-name}} --baud 460800 -s 8mb && sleep 2 && espflash monitor
+	espflash flash \
+		--erase-parts nvs \
+		--partition-table partitions.csv \
+		--flash-size ${FLASHSIZE} \
+		--baud ${BAUDRATE} \
+		target/xtensa-esp32-espidf/release/{{project-name}} && sleep 2 && espflash monitor --baud ${BAUDRATE}
```

Comment:
nit: I would use the default monitor baudrate

Reply:
I was intermittently getting Timeouts connecting to the older MCU board using the default baud rate. It would connect 25% successfully of the time. Setting the baudrate to a lower value, in my case, 115200, increased the successful connections.

In the Makefile, I set a default value for the Baud rate to 460800.  The user can change / reduce the value, depending on their board capabilities.